p2p/discover/discfilter/discfilter.go

```diff
@@ -1,44 +1,42 @@
 package discfilter
 
 import (
+	"time"
+
 	lru "github.com/hashicorp/golang-lru"
+	"github.com/zekroTJA/timedmap"
 
 	"github.com/ethereum/go-ethereum/p2p/enode"
-	"github.com/ethereum/go-ethereum/p2p/enr"
 )
 
 var (
-	enabled    = false
-	dynamic, _ = lru.New(50000)
+	enabled         = false
+	cleanupInterval = 5 * time.Minute // check every 5 minute to remove the expired entries
+	dynamic         = timedmap.New(cleanupInterval)
+	static, _       = lru.New(50000)
 )
 
 func Enable() {
 	enabled = true
 }
 
-func Ban(id enode.ID) {
-	if enabled {
-		dynamic.Add(id, struct{}{})
-	}
-}
-
-func BannedDynamic(id enode.ID) bool {
+func BanDynamic(id enode.ID, banInterval time.Duration) {
```

Comment:
I don't see this new function being used other than in the test, is that intentional?